### ISSUE TITLE
Compilation error on OCaml 5.0

### ISSUE THREAD

**Issue by pklehre** [hbr/fmlib#5]
I get the following compilation error when trying to install fmlib on the 5.0.0~alpha1+flambda opam switch

```
pkl@zeta ocamlec]$ opam install fmlib
The following actions will be performed:
  ∗ install fmlib_parse 0.5.1 [required by fmlib]
  ∗ install fmlib       0.5.1
===== ∗ 2 =====
Do you want to continue? [Y/n] y

<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
⬇ retrieved fmlib.0.5.1  (cached)
⬇ retrieved fmlib_parse.0.5.1  (cached)
[ERROR] The compilation of fmlib_parse.0.5.1 failed at "dune build -p fmlib_parse -j 4 @install".

#=== ERROR while compiling fmlib_parse.0.5.1 ==================================#
# context     2.1.2 | linux/x86_64 | ocaml-option-flambda.1 ocaml-variants.5.0.0~alpha1+options | https://opam.ocaml.org/#adaae788
# path        ~/.opam/5.0.0~alpha1+flambda/.opam-switch/build/fmlib_parse.0.5.1
# command     ~/.opam/opam-init/hooks/sandbox.sh build dune build -p fmlib_parse -j 4 @install
# exit-code   1
# env-file    ~/.opam/log/fmlib_parse-80944-1b5544.env
# output-file ~/.opam/log/fmlib_parse-80944-1b5544.out
### output ###
# 91 |     val run_on_stream: char Stream.t -> t -> doc
# [...]
# Error: Unbound module Stream
# (cd _build/default && /home/pkl/.opam/5.0.0~alpha1+flambda/bin/ocamlc.opt -w -40 -g -bin-annot -I src/parse/.fmlib_parse.objs/byte -I /home/pkl/.opam/5.0.0~alpha1+flambda/lib/base -I /home/pkl/.opam/5.0.0~alpha1+flambda/lib/base/base_internalhash_types -I /home/pkl/.opam/5.0.0~alpha1+flambda/lib/base/caml -I /home/pkl/.opam/5.0.0~alpha1+flambda/lib/base/shadow_stdlib -I /home/pkl/.opam/5.0.0~[...]
# File "src/parse/interfaces.mli", line 37, characters 29-37:
# 37 |     val run_on_stream: token Stream.t -> t -> t
#                                   ^^^^^^^^
# Error: Unbound module Stream
# (cd _build/default && /home/pkl/.opam/5.0.0~alpha1+flambda/bin/ocamlc.opt -w -40 -g -bin-annot -I src/parse/.fmlib_parse.objs/byte -I /home/pkl/.opam/5.0.0~alpha1+flambda/lib/base -I /home/pkl/.opam/5.0.0~alpha1+flambda/lib/base/base_internalhash_types -I /home/pkl/.opam/5.0.0~alpha1+flambda/lib/base/caml -I /home/pkl/.opam/5.0.0~alpha1+flambda/lib/base/shadow_stdlib -I /home/pkl/.opam/5.0.0~[...]
# File "src/parse/source_extractor.mli", line 19, characters 24-32:
# 19 | val run_on_stream: char Stream.t -> t -> t
#                              ^^^^^^^^
# Error: Unbound module Stream



<><> Error report <><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
┌─ The following actions failed
│ λ build fmlib_parse 0.5.1
└─ 
╶─ No changes have been performed

```

**Comment by hbr** [comment]
This is a strange compilation error. The compiler complains about an unknown module `Stream`. However the module `Stream` is part of ocaml's standard library. Is it possible that the alpha version of ocaml 5.0.0 doesn't have a complete standard library?

Regards
Helmut

**Comment by pklehre** [comment]
It could be an OCaml 5.0 bug. Would you like to post an issue here? https://github.com/ocaml/ocaml/issues

**Comment by pklehre** [comment]
I obtained the same error message when trying to install mariadb, so I will post it on the ocaml issues page.

**Comment by pklehre** [comment]
It appears that Stream has indeed been removed: https://github.com/ocaml/ocaml/pull/10896

**Comment by hbr** [comment]
Hello Kristian,

I have learned that the module `Stream` has been deprecated from 4.14 on and will definitely **not** be available in 5.0. The module `Stream` is part of the library `camlp-streams` which you can install from opam. You can install it as a workaround and the compilation should work.

In the future I will either remove the dependency on `Stream` or make the dependency on `camlp-streams` explicit.

However thanks for reporting the issue. I had not been aware of the removal of streams.

Helmut
